core/runtime/execute_engine.cpp

```diff
@@ -115,11 +115,15 @@ std::vector<at::Tensor> execute_engine(std::vector<at::Tensor> inputs, c10::intr
   }
 
   // Whether cudagraphs needs to record the graph on this pass
-  bool need_cudagraphs_record = (CUDAGRAPHS_MODE && (!_cudagraphs_validate_shapes(inputs, compiled_engine)));
+  bool need_cudagraphs_record =
```

Comment:
Seems like we need something a bit more comprehensive than a bunch of booleans for cudagraph rerecord

Reply:
I agree with it. I'm thinking of below structure. Could you review [PR3276]?(https://github.com/pytorch/TensorRT/pull/3276)
https://github.com/pytorch/TensorRT/pull/3276/files#diff-3d6304b0f21f64bdf7867ceac346866d505619401e9c7b6a696cfe3bb567254eR37-R63